client/core/trade.go

```diff
@@ -1074,21 +1050,10 @@ func (t *trackedTrade) revokeMatch(matchID order.MatchID, fromServer bool) error
 	}
 
 	// Notify the user of the failed match.
-	corder, _ := t.coreOrderInternal() // no cancel order
+	corder := t.coreOrderInternal() // no cancel order
 	t.notify(newOrderNote("Match revoked", fmt.Sprintf("Match %s has been revoked",
 		token(matchID[:])), db.WarningLevel, corder))
 
-	// Send out a data notification with the revoke information.
-	cancelOrder := &Order{
-		Host:     t.dc.acct.host,
-		MarketID: t.mktID,
-		Type:     order.CancelOrderType,
-		Stamp:    encode.UnixMilliU(time.Now()),
-		Epoch:    corder.Epoch,
-		TargetID: corder.ID, // the important part for the frontend
-	}
-	t.notify(newOrderNote("revoke", "", db.Data, cancelOrder))
-
```

Comment:
This is a bug, I think. The "revoke" subject data message is interpreted as a revoked order on the front end, not a revoked match.

Reply:
Right. I'd mentioned this before. The logged note looked like an order was revoked. Which reminds me, we still don't have a DEX-originating `revoke_order` route yet. Clients have to "wait" for reconnect to know that orders have been revoked.

Reply:
I brought up the lack of `revoke_order` in https://github.com/decred/dcrdex/pull/732.

This is the case where a revoked match results in the order revoking too, but only when the user was at fault.

Reply:
This doesn't actually revoke the order, actually I'm not sure any match revocation triggers order revocation even for at-fault clients, it's just a notification that appears to mean that an order was revoked when it actually hasn't been revoked.

Reply:
But `Swapper` does revoke the order for the at-fault party via the `unbookHook`.

Reply:
OK, so plan is to implement `revoke_order`.  This will facilitate both (1) the at-fault order being revoked when the match is revoked (the relevant case here), and (2) revoking an order when it becomes unexpectedly unfunded before fill (the case in https://github.com/decred/dcrdex/pull/732).